flight/search/plot_algo.py

```diff
@@ -0,0 +1,601 @@
+"""
+The actual algorithm powering the drone's navigation
+"""
+
+from math import floor, sqrt
+from bisect import insort
+from pathfinding.finder.a_star import AStarFinder
+from pathfinding.core.grid import Grid, Node
+from numpy import zeros, int8
+from numpy.typing import NDArray
+from cell_map import CellMap
+from cell import Cell
+from helper import AIR_DROP_AREA
+from segmenter import segment, rotate_shape, SUAS_2023_THETA
+
+
+class Compressor:
+    """
+    Compresses the search area into a low resolution graph based on the drone's
+    search area, exponentially reducing the time the algorithm takes
+
+    Methods
+    -------
+    analyze_cell(i : int, j: int, s: int, cell_map: CellMap) -> int
+        Returns the number of in-bounds cells
+    get_circum_square_r(r: int) -> int
+        returns the side length of the square inside the view radius circle
+    __init_compressed_grid(cell_size : int, cell_map : CellMap) -> NDArray:
+        Returns empty compressed grid with correct dimensions
+    compress(radius: int, cell_map: CellMap) -> list[list[tuple[bool, int, int]]]
+        compresses the given cell_map into larger cells based on the given radius
+    """
+
+    @staticmethod
+    def analyze_cell(i: int, j: int, size: int, cell_map: CellMap) -> int:
+        """
+        Given the compressed index, returns the number of valid locations
+        within the cell
+
+        Parameters
+        ----------
+        i : int
+            the row of the compressed cell
+        j : int
+            the column of the compressed cell
+        size: int
+            size of the cell
+        cell_map : CellMap
+            the map being compressed
+
+        Returns
+        -------
+        value : int
+            the number of valid cells within the compressed cell
+        """
+        score: int = 0
+        row_start: int = size * i
+        row_end: int = min((size * (i + 1) - 1), len(cell_map.data) - 1)
+        col_start: int = size * j
+        col_end: int = min((size * (j + 1) - 1), len(cell_map[0]) - 1)
+
+        for row in range(row_start, row_end + 1):
+            for col in range(col_start, col_end + 1):
+                try:
+                    if cell_map[row][col].is_valid:
+                        score += 1
+                except IndexError:
+                    pass
```

Comment:
Not sure when there would be a index error, but you should try to avoid that by constraining your `row_end` and `col_end` vars rather than a try-except